crates/sui-framework/packages/dwallet-system/sources/ethereum_state.move

```diff
@@ -79,6 +83,7 @@ module dwallet_system::ethereum_state {
             id: object::new(ctx),
             data,
             time_slot,
+            state_root
         };
 
         if (new_state.time_slot > latest_ethereum_state.last_slot) {
```

Comment:
This should be a restrictive check: you don't allow creation of the `EthereumState` objects unless they are newer than the latest

Reply:
Also, you must verify they are both in the same network.

Reply:
I moved the check of the time slot so that if the time slot is not bigger than the latest it would not create the state.
Furthermore, there is a check for that inside the state verification function.

Also, as we saw together on last Thursday, the verification would fail if the network is not the same network, as this is checked when validating the sync committee signatures. 
The network is defined inside the `ConsensusStateManager` object that is serialized into the `data` field of this object, and it's configuration is hard-coded into helios.